lnrpc/routerrpc/router_server.go

```diff
@@ -740,10 +744,57 @@ func (s *Server) trackPayment(identifier lntypes.Hash,
 	case err != nil:
 		return err
 	}
+
+	// Stream updates to the client
+	return trackPaymentStream(
+		s,
+		router,
+		subscription,
+		noInflightUpdates,
+		stream.Send,
+		stream.Context,
+		fmt.Sprintf("Payment stream %v canceled", identifier),
+	)
+}
+
+// TrackPayments returns a stream of payment state updates.
+func (s *Server) TrackPayments(request *TrackPaymentsRequest,
+	stream Router_TrackPaymentsServer) error {
+
+	log.Debug("TrackPayments called")
+
+	router := s.cfg.RouterBackend
+
+	// Subscribe to payments.
+	subscription, err := router.Tower.SubscribeAllPayments()
+	if err != nil {
+		return err
+	}
+
+	// Stream updates to the client
+	return trackPaymentStream(
+		s,
+		router,
+		subscription,
+		request.NoInflightUpdates,
+		stream.Send,
+		stream.Context,
+		"Payment stream canceled",
+	)
+}
+
+// trackPaymentStream streams payment updates to the client.
+func trackPaymentStream(s *Server,
```

Comment:
Maybe if you make this a method of `Server`, you don't need to pass `s` and `router`.

Reply:
Good, done